frontend/packages/db-structure/src/parser/drizzle/parser.ts

```diff
@@ -0,0 +1,1166 @@
+import * as ts from 'typescript'
+import type {
+  Column,
+  Columns,
+  Constraints,
+  ForeignKeyConstraint,
+  Index,
+  Table,
+} from '../../schema/index.js'
+import type { Processor, ProcessResult } from '../types.js'
+import {
+  convertDefaultValue,
+  convertDrizzleTypeToPgType,
+  convertReferenceOption,
+} from './convertToPgType.js'
+
+interface DrizzleTableDefinition {
+  name: string
+  columns: Record<string, DrizzleColumnDefinition>
+  indexes: Record<string, DrizzleIndexDefinition>
+  compositePrimaryKey?: CompositePrimaryKeyDefinition
+  comment?: string | undefined
+}
+
+interface DrizzleColumnDefinition {
```

Comment:
It would be better if we could use the type definitions provided by drizzle!

Reply:
I have tried a few things about the above, but I am currently using `@swc/core` for AST analysis and found it a bit difficult.😞